src/main.py

```diff
@@ -70,7 +58,8 @@ async def on_list(
     ),
 ) -> None:
     """Download and list all media sent in a chosen text channel."""
-    bc = get_controller(interaction.guild_id)
+    global controllers
```

Comment:
One only needs to do `global X` if they're changing the object that `X` points to. In this function we're only modifying what `controllers` contains, not actually re-assigning it (`controllers =`).

So the existing `global controller` and this one should be unnecessary.

Likewise for the other functions.

Reply:
TIL